// advanced_knx/dist/RawMod/CustomMessageHandlerTemplates.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _KnxConstants = _interopRequireDefault(require("../KnxConstants"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**************************************************
 * This file contains some custom message handler *
 **************************************************/

/*
 * 'CustomMessageHandlerTemplates' contains custom message handler templates for different types of messages
 * Every template consists of a register template (used to register the handler)
 */
const CustomMessageHandlerTemplates = {
  ncdNackHandlerTemplate: (sender, receiver) => {
    return {
      cemi: {
        src_addr: sender,
        dest_addr: receiver,
        apdu: {
          tpci: _KnxConstants.default.KNX_TPCI_TYPES.TPCI_NCD | _KnxConstants.default.KNX_TPCI_SUBTYPES.TPCI_NCD_NACK
        }
      }
    };
  },
  ncdAckHandlerTemplate: (sender, receiver) => {
    return {
      cemi: {
        src_addr: sender,
        dest_addr: receiver,
        apdu: {
          tpci: _KnxConstants.default.KNX_TPCI_TYPES.TPCI_NCD | _KnxConstants.default.KNX_TPCI_SUBTYPES.TPCI_NCD_ACK
        }
      }
    };
  },
  memoryResponseTemplate: (sender, receiver) => {
    return {
      cemi: {
        src_addr: sender,
        dest_addr: receiver,
        apdu: {
          apci: 'Memory_Response'
        }
      }
    };
  },
  adcResponseTemplate: (sender, receiver) => {
    return {
      cemi: {
        src_addr: sender,
        dest_addr: receiver,
        apdu: {
          apci: 'ADC_Response'
        }
      }
    };
  },
  devDescrResponseTemplate: (sender, receiver) => {
    return {
      cemi: {
        src_addr: sender,
        dest_addr: receiver,
        apdu: {
          apci: 'DeviceDescriptor_Response'
        }
      }
    };
  },
  propertyValueResponseTemplate: (sender, receiver) => {
    return {
      cemi: {
        src_addr: sender,
        dest_addr: receiver,
        apdu: {
          apci: 'PropertyValue_Response',
          tpci: _KnxConstants.default.KNX_TPCI_TYPES.TPCI_NDP
        }
      }
    };
  }
};
var _default = CustomMessageHandlerTemplates;
exports.default = _default;
